common/lib/common-definitions/src/logger.ts

```diff
@@ -33,13 +38,27 @@ export interface ITelemetryProperties {
     [index: string]: TelemetryEventPropertyType | ITaggedTelemetryPropertyType;
 }
 
+/**
+ * Set of properties defined for the base logger to use only primitive types.
+ */
+export type TelemetryBasePropertyType = string | number | boolean | undefined;
+
+export interface ITaggedBasePropertyType {
```

Comment:
What do you think about dropping the explicit `ITagged` interfaces and instead defining this helper type:

```typescript
export type TaggedProperty<TValue> = {
    value: TValue,
    tag: string,
};
```

Then `ITelemetryProperties` can do `TelemetryEventPropertyType | TaggedProperty<TelemetryEventPropertyType>` (and likewise for `ITelemetryBaseProperties`)